plutus-core/untyped-plutus-core/test/Transform/caseOfCase2.plc.golden

```diff
@@ -0,0 +1,5 @@
+(case
+  [ [ [ (force (builtin ifThenElse)) b_0 ] t_1 ] (constr 1) ]
+  (con integer 1)
+  (con integer 2)
+)
```

Comment:
Worth a comment on why we don't transform

```haskell
case (if b then t else False) of
     True  -> 1
     False -> 2
```

into

```haskell
if b
   then case t of
        True  -> 1
        False -> 2
   else 2
```

I assume this is because we don't want to duplicate branches? (and what we do right now does duplicate them, but since all constants are now, the duplication get removed by a subsequent pass of case-of-known-constructor or whatever).

Reply:
Added a comment:

```
-- Unless both branches are known constructors, the case-of-case transformation
-- may increase the program size.
```